publication-commons/src/test/java/no/unit/nva/publication/permission/strategy/EditorPermissionStrategyTest.java

```diff
@@ -0,0 +1,101 @@
+
+    @ParameterizedTest(name = "Should allow editor {0} operation on non-degree resources")
+    @EnumSource(value = PublicationOperation.class, mode = Mode.EXCLUDE, names = {"DELETE", "TICKET_PUBLISH"})
+    void shouldAllowEditorOnNonDegreeBasedOnOwner(PublicationOperation operation)
+        throws JsonProcessingException, UnauthorizedException {
+
+        var institution = randomUri();
+        var resourceOwner = randomString();
+        var editorUsername = randomString();
+        var cristinId = randomUri();
+
+        var requestInfo = createUserRequestInfo(editorUsername, institution, getAccessRightsForEditor(),
+                                                cristinId, randomUri());
+        var publication =
+            createNonDegreePublication(resourceOwner, institution).copy()
+                .withStatus(PublicationOperation.UNPUBLISH == operation ? PUBLISHED : UNPUBLISHED)
+                .build();
+        var userInstance = RequestUtil.createUserInstanceFromRequest(requestInfo, identityServiceClient);
+
+        Assertions.assertTrue(PublicationPermissionStrategy
+                                  .create(publication, userInstance, uriRetriever)
+                                  .allowsAction(operation));
+    }
+
+    @ParameterizedTest(name = "Should deny editor {0} operation on non-degree resources")
+    @EnumSource(value = PublicationOperation.class, mode = Mode.EXCLUDE, names = {"UNPUBLISH", "UPDATE", "TERMINATE"})
+    void shouldDenyEditorOnNonDegree(PublicationOperation operation)
+        throws JsonProcessingException, UnauthorizedException {
+
+        var institution = randomUri();
+        var resourceOwner = randomString();
+        var editorUsername = randomString();
+        var cristinId = randomUri();
+
+        var requestInfo = createUserRequestInfo(editorUsername, institution, getAccessRightsForEditor(),
+                                                cristinId, randomUri());
+        var publication = createNonDegreePublication(resourceOwner, institution);
+        var userInstance = RequestUtil.createUserInstanceFromRequest(requestInfo, identityServiceClient);
+
+        Assertions.assertFalse(PublicationPermissionStrategy
+                                   .create(publication, userInstance, uriRetriever)
+                                   .allowsAction(operation));
+    }
+
+    @Test
+    void shouldGiveEditorPermissionToUnpublishPublicationWhenPublicationIsFromTheirInstitution()
+        throws JsonProcessingException, UnauthorizedException {
+
+        var editorName = randomString();
+        var editorInstitution = randomUri();
+        var resourceOwner = randomString();
+        var cristinId = randomUri();
+
+        var requestInfo = createUserRequestInfo(editorName, editorInstitution, getAccessRightsForEditor(),
+                                                cristinId, randomUri());
+        var publication = createPublication(resourceOwner, editorInstitution, cristinId);
```

Comment:
This publication contains a published file, which is very essential to the permission check (in generall, not editor specifically). An editor (and curator)  should be able to unpublish whether or not the publication has any published files. Should we make an explicit test for this? As it goes kind of unnoticed now.
Up to you.